### DISCUSSION TITLE
Cannot find browser.js module

### DISCUSSION THREAD

**Discussion by Baspa** [spatie/browsershot#769] [Bugs]
## Error

I'm currently struggling with getting Pupeteer back to work. I'm sure it worked before but somehow it doesn't work anymore. I installed my application on a Laravel Forge server. I followed the [installation instructions](https://spatie.be/docs/browsershot/v2/requirements#content-installing-puppeteer-a-forge-provisioned-server) on the website. This threw some errors which I managed to fix using [this answer](https://github.com/spatie/browsershot/discussions/681#discussioncomment-5561589).

On my Laravel Forge server I am still, getting the following error:

```
Output:
================


Error Output:
================
node:internal/modules/cjs/loader:959
  throw err;
  ^

Error: Cannot find module '/home/forge/workoutplanner.so/vendor/spatie/browsershot/bin/browser.js'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)
    at Function.Module._load (node:internal/modules/cjs/loader:804:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

When I try to locate the file I don't find a `browser.js` in the directory, but **do** find a `browser.cjs`. I am not familiar with the extension so I'm not sure if that is the cause that the module cannot be found. I already tried a new `npm install` and already installed puppeteer globally. 

## Package versions

Puppeteer version: ^17.0.0
Browsershot version: ^3.57

## Code

The code that causes the error: 

```php
// @see https://github.com/spatie/browsershot/discussions/681#discussioncomment-5561589 
Browsershot::html($html)
    ->setChromePath('/snap/bin/chromium')
    // ->setIncludePath('$PATH:/usr/local/bin')
    ->format('A4')
    ->savePdf($path . '/' . $filename);
```

**Comment by hyphen81** [comment]
Did you ever find a resolution to this? I'm running into the same error.

**Reply by Baspa** [reply]
I've got to admit that I didn't have any time to check on this lately.. So I don't have a solution (yet). Do you? @hyphen81

**Reply by hyphen81** [reply]
I ended up just renaming the file and everything worked. Kind of a hack, but after all the time I spent trying to find an answer, I'm happy to just be moving on to other things.

https://stackoverflow.com/questions/77379878/browsershot-module-not-found-browser-js?noredirect=1#comment136416269_77379878

**Reply by Baspa** [reply]
Thank you very much! I'm gonna take a look at it later today. Hopefully there will be a solid, definitive solution for this.

**Reply by hyphen81** [reply]
I will say, the problem I just discovered with this hack, is that i now have to rename the file in production every time I deploy code because it gets overwritten on build. So that really sucks. I wish someone had a real solution to this.

**Reply by Baspa** [reply]
Believe it or not but adding this to my deploy script did the trick:

`php artisan queue:restart`

Seems like my queue was never restarted after a deploy.. @hyphen81
